cfme/tests/openstack/cloud/test_volumes.py

```diff
@@ -0,0 +1,189 @@
+
+    Instance(instance_name, provider).load_details()
+    toolbar.select('Configuration', 'Attach a Cloud Volume to this Instance')
+    assert len(select_volume.all_options) > 0
+
+    select_volume.select_by_value(select_volume.get_value_by_text(volume_name))
+    pytest_selenium.send_keys(device_input, '/dev/vdb')
+    pytest_selenium.click(submit_changes)
+    assert_message_contain('Attaching Cloud Volume')
+
+    wait_for(provider.is_refreshed, [None, 10], delay=5)
+    info_el = InfoBlock.element('Relationships', 'Cloud Volumes')
+    pytest_selenium.click(info_el)
+    params = {'Name': volume_name, 'Status': 'in-use'}
+    res = list_tbl.find_rows_by_cells(params)
+    assert res, 'Volume does not appear in instance relationships'
+
+
+@pytest.mark.uncollectif(lambda: PROD_VERSION > '4.1')
+def test_detach_volume_from_instance_page(provider):
+    """Detaches volume from instance from Instance page"""
+    vm_name = None
+    # Find an instance with attached volumes
+    for instance in provider.mgmt._get_all_instances():
+        info = instance.to_dict()
+        if info['os-extended-volumes:volumes_attached']:
+            vm_name = info['name']
+            break
+    assert vm_name, "Can't find any instance with attached volume"
+
+    Instance(vm_name, provider).load_details()
+    toolbar.select('Configuration', 'Detach a Cloud Volume from this Instance')
+    v_option = choice(select_volume.all_options[1:])
+    select_volume.select_by_value(v_option[1])
+    pytest_selenium.click(submit_changes)
+    assert_message_contain('Detaching Cloud Volume')
+
+    wait_for(provider.is_refreshed, [None, 10], delay=5)
+    info_el = InfoBlock.element('Relationships', 'Cloud Volumes')
+    pytest_selenium.click(info_el)
+    res = list_tbl.find_cell('Name', v_option[0])
+    assert not res, 'Volume does not disappear from instance relationships'
+
+
+@pytest.mark.uncollectif(lambda: PROD_VERSION > '4.1')
+def test_attach_volume_from_volume_page(provider):
+    """Attaches pre-created volume to an instance from Volume page"""
+    vms = filter(lambda vm: vm.power_state == 'ACTIVE', provider.mgmt.all_vms())
+    inst_name = choice(vms).name
+    navigate_to(Volume, 'All')
+    params = {'Status': 'available',
+              'Cloud Provider': provider.get_yaml_data()['name']}
+    list_tbl.click_row_by_cells(params, 'Name')
+    vname = get_volume_name()
+    toolbar.select('Configuration', 'Attach this Cloud Volume to an Instance')
+    assert len(select_vm.all_options) > 0
+
+    select_vm.select_by_value(select_vm.get_value_by_text(inst_name))
+    pytest_selenium.send_keys(device_input, '/dev/vdb')
+    pytest_selenium.click(attach_volume)
+    assert_message_contain('Attaching Cloud Volume')
+
+    wait_for(provider.is_refreshed, [None, 10], delay=5)
+    navigate_to(Volume, 'All')
+    params = {'Name': vname, 'Status': 'in-use'}
+    res = list_tbl.find_rows_by_cells(params)
+    assert res, 'Volume does not marked as in-use'
```

Comment:
This is a good usage of an assert message since the result in re might not be entirely clear.